tf2_filter/include/tf2_filter/tf2_filter.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2018 Robert Bosch GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef TF2_FILTER__TF2_FILTER_HPP_
+#define TF2_FILTER__TF2_FILTER_HPP_
+
+#include <map>
+#include <unordered_set>
+#include <string>
+#include "tf2_filter/visibility_control.h"
+#include <tf2_msgs/msg/tf_message.hpp>
+
+namespace tf2_filter
+{
+typedef std::unordered_set<std::string> TargetSet;
+typedef std::map<std::string, TargetSet> FilterMap;
```

Comment:
By doing the checking as a 2-step process, it would consume fewer resources for the common case where the parent is already not part of the chain.

This may be an over-optimization, but it wasn't hard to do ;-)

Reply:
I'm worried about it being fragile. The tree may be reconfigured and requiring knowing the topology of the tree to set these settings would make this setting fragile even if it's still completely connected.

Reply:
Okay, I'll replace it with a plain set.